client/src/util/authentication.ts

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable no-console */
+import { DateTime } from 'luxon';
+import axios from 'axios';
+
+// Used to get authCode from url
+export function getCodeParam(): string | null {
+    const queryString: string = window.location.search;
+    const urlParams: URLSearchParams = new URLSearchParams(queryString);
+    const codeParam: string | null = urlParams.get('code');
+  
+    return codeParam;
+};
+
+export function getGitLabAccessCode(
+    clientId: string | undefined,
+    requestedScopes: string | undefined,
+    redirectUri: string | undefined,
+): void {
+    if (clientId !== undefined && requestedScopes !== undefined && redirectUri !== undefined) {
+        window.location.assign(
+            `https://gitlab.com/oauth/authorize?client_id=${clientId}&redirect_uri=${redirectUri}&response_type=code&scope=${requestedScopes}`,
+        );
+    }
+    else {
+        console.log('Input error');
```

Comment:
if you expect errors, [throw](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw) an exception and put the function call in a try....catch block later.